base/error.jl

```diff
@@ -197,15 +197,16 @@ windowserror(p, code::UInt32=Libc.GetLastError(); extrainfo=nothing) = throw(Mai
 """
     @assert cond [text]
 
-Throw an [`AssertionError`](@ref) if `cond` is `false`. Preferred syntax for writing assertions.
+Throw an [`AssertionError`](@ref) if `cond` is `false`. Debugging tool for writing assertions.
```

Comment:
```suggestion
Throw an [`AssertionError`](@ref) if `cond` is `false`.
```
I think it is pretty clear that `@assert` is for writing assertions based on the syntax, and especially given the first sentence.

Reply:
My preoccupation about this doc entry is that from many discussions (at Discourse) I came to the conclusion that one should not use `@assert condition` as a substitute for `if !condition throw(error()) end`, because `@assert` can in some circumstances by eliminated by the compiler at runtime. This must be very, very clear in the docs, or can cause correctness problems in people's code. Since I have a bare understanding of how it works and its actual utility, I cannot really discuss this in further detail. From my point of view I just won't use it anymore, ever, in production code. 

For instance, at this point the phrase **Throw an [AssertionError](@ref) if `cond` is `false** is only partially correct, as this may fail in the situation where assert is removed by the compiler. I am not in favor of relying on the good understanding of a previous sentence for a user to avoid this problem. Thus, I would be pedantic in this sense. For power users, nothing really matters, it is not for them that docs must be obsessively clear.

Reply:
That makes sense. Thank you for your explanation. I agree.